### PR TITLE
Incorporate quanTIseq into immune-deconv module

### PR DESCRIPTION
**Pull request by sjspielman**
#### What scientific question is your analysis addressing?

As discussed in issue #1238, this PR adds the `quanTIseq` method to the `immune-deconv` module.

#### What was your approach?

Slight modification of existing script to run deconvolution to export `RDS` files instead of `.RData` (for tracking different output names per method). We now call the deconvolution script twice from the analysis shell script, once per method. We also add a quick check into the R script to convert any provided method to all lowercase, which `immunedeconv` requires.

#### What GitHub issue does your pull request address?
#1238 

#### What types of results are included (e.g., table, figure)?

Updated result files as RDS:

```
results/xcell_deconv-output.rds
results/quantiseq_deconv-output.rds
```

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/immune-deconv/01-immune-deconv.R [hunk 1]
```diff
@@ -87,13 +91,11 @@ deconv <- function(expr_input, clin_file, method) {
...
 combo <- expand.grid(expr_input, deconv_method, stringsAsFactors = F) 
 deconv_output <- apply(combo, 1, FUN = function(x) deconv(expr_input = x[1], clin_file = clin_file, method = x[2]))
 deconv_output <- do.call(rbind.data.frame, deconv_output)
 
-# save output to RData object
-print("Writing output to file..")
-save(deconv_output, file = output_file)
-print("Done!")
+# save output to RDS file
+write_rds(deconv_output, output_file)
```
Comment (jaclyn-taroni) [inline analyses/immune-deconv/01-immune-deconv.R:101]: To check my understanding - the `deconvolute()` steps are run for each dataset separately, but then we're putting them into the same data frame. For xCell, the scores don't change depending on whether the data are together or not (see discussion on #292). I don't know that the same is true of `quantiseq`, but what I'm actually getting at is that our result almost implies that we ran them together when that's not the case. I think this (rambling) comment could be addressed by adding a column in the resulting data frame that keeps track of which input dataset the sample comes from.

File: analyses/immune-deconv/README.md [hunk 2]
```diff
@@ -18,6 +21,9 @@ xCell outputs immune scores as arbitrary scores that represent cell type abundan
 Importantly, these scores may be compared between samples (inter-sample comparisons), but _may not_ be compared across cell types or cancer types, as described in the [`immunedeconv` documentation](https://icbi-lab.github.io/immunedeconv/articles/immunedeconv.html#interpretation-of-scores). This is in part because xCell is actually a signature-based method and not a deconvolution method, as is described in the [xCell Publication](https://doi.org/10.1186/s13059-017-1349-1):
 > Unlike signature-based methods, which output independent enrichment scores per cell type, the output from deconvolution-based methods is the inferred proportions of the cell types in the mixture.
 
+Therefore, we also use `quanTIseq` as a complementary method. Although `quanTIseq` looks at fewer cell types, the scores can be interpreted as absolute fractions, thereby allowing comparison _both_ across samples and cell types.
```
Comment (jaclyn-taroni) [inline analyses/immune-deconv/README.md:24]: I'd link the documentation on interpretation of scores again for good measure.

**Review by jaclyn-taroni — COMMENTED** [review]
The changes you're adding are very reasonable! The one comment that is preventing me from approving yet is about whether we should go a touch furthe.

File: analyses/immune-deconv/01-immune-deconv.R (hunk 1 above)
Comment (sjspielman) [reply analyses/immune-deconv/01-immune-deconv.R:101]: :100:

**Review by jaclyn-taroni — APPROVED** [review]
👍🏻

**Pull request merged**